### ISSUE TITLE
Frequent nodejs error: maxlistenersexceeded

### ISSUE THREAD

**Issue by jeremiahpslewis** [garden-io/garden#2422]
### Current Behavior

Error occurs when executing garden run task
```bash
(node:1318) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 drain listeners added to [ReadEntry]. Use emitter.setMaxListeners() to increase limit
```

### Expected behavior

No error.

### Your environment

* **OS:** Apple Silicon macOS 11.4
* **How I'm running Kubernetes:** Docker for Mac

`garden version`
0.12.22

**Comment by eysi09** [comment]
Hi @jeremiahpslewis 

We haven't seen this on our end before. Anything particular in your set up that you think could explain this? And if you have a reproducible, that would be fantastic.

**Comment by jeremiahpslewis** [comment]
This happens when I try to export too many / too large files to artifacts from a container. Don't have a clean reproducible at the moment, but the project where this frequently comes up is open source: https://github.com/jeremiahpslewis/alec

**Comment by edvald** [comment]
This looks like a configuration issue somewhere in our code that triggers this warning (11 listeners on an event emitter could well be totally normal). Does something actually fail, or just the annoying log message? Also, what type of task are you running?

**Comment by jeremiahpslewis** [comment]
It fails entirely, interrupts the workflow.

**Comment by jeremiahpslewis** [comment]
Task generates a synthetic dataset and then writes it to file, it's then supposed to export the files as an artifact (lots of moderately sized objects).

**Comment by edvald** [comment]
The error doesn't indicate a crash per se, so we'd probably need a bit more info to go on. Is there something in `error.log` or some other surrounding log lines that could indicate something? Also, might help to run with `--log-level=silly` to see if there's something suspicious around the time of crash.

**Comment by thsig** [comment]
Closed by #4258.
